core/templates/domain/skill/skill-backend-api.service.ts

```diff
@@ -16,124 +16,176 @@
  * @fileoverview Service to send changes to a skill to the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-require('domain/skill/skill-domain.constants.ajs.ts');
-
-angular.module('oppia').factory('SkillBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService',
-  'EDITABLE_SKILL_DATA_URL_TEMPLATE', 'SKILL_DATA_URL_TEMPLATE',
-  function(
-      $http, $q, UrlInterpolationService,
-      EDITABLE_SKILL_DATA_URL_TEMPLATE, SKILL_DATA_URL_TEMPLATE) {
-    var _fetchSkill = function(skillId, successCallback, errorCallback) {
-      var skillDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_SKILL_DATA_URL_TEMPLATE, {
-          skill_id: skillId
-        });
-
-      $http.get(skillDataUrl).then(function(response) {
-        var skill = angular.copy(response.data.skill);
-        var groupedSkillSummaryDicts = angular.copy(
-          response.data.grouped_skill_summaries);
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { SkillDomainConstants } from 'domain/skill/skill-domain.constants';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service.ts';
+import cloneDeep from 'lodash/cloneDeep';
+
+
+export interface ISkillUpdatePayload {
+  version: number,
+  commitMessage: string,
+  changeDicts: Array<IChangeDict>
+}
+
+export interface IChangeDict {
+  cmd?: string;
+  propertyName?: string;
+  oldValue?: string;
+  newValue?: string;
+}
+
+export interface ISkillResponse {
+  skill?: Object;
```

Comment:
Why is everything mark optional in the response type?

Reply:
I'm using the same interface for all functions and the properties hence vary. Should I make different interfaces for each function?

Reply:
Yes you should.

Reply:
Done